src/ui/public/parse_query/parse_query.js

```diff
@@ -1,27 +1,51 @@
-import { toUser } from 'ui/parse_query/lib/to_user';
-import { ParseQueryLibFromUserProvider } from 'ui/parse_query/lib/from_user';
-
-import { uiModules } from 'ui/modules';
-uiModules
-  .get('kibana')
-  .directive('parseQuery', function (Private) {
-    const fromUser = Private(ParseQueryLibFromUserProvider);
-
-    return {
-      restrict: 'A',
-      require: 'ngModel',
-      scope: {
-        'ngModel': '='
-      },
-      link: function ($scope, elem, attr, ngModel) {
-        const init = function () {
-          $scope.ngModel = fromUser($scope.ngModel);
-        };
-
-        ngModel.$parsers.push(fromUser);
-        ngModel.$formatters.push(toUser);
-
-        init();
-      }
-    };
-  });
+import _ from 'lodash';
+import angular from 'angular';
+
+export function getDefaultQuery() {
+  return { match_all: {} };
+}
+
+export function isDefaultQuery(query) {
+  return _.isEqual(query, getDefaultQuery());
+}
+
+export function getTextQuery(query) {
+  return {
+    query_string: { query }
+  };
+}
+
+export function isTextQuery(query) {
+  return _.has(query, 'query_string');
+}
+
+export function getQueryText(query) {
+  return _.get(query, ['query_string', 'query']) || '';
+}
+
+export function parseQuery(query) {
+  if (!_.isString(query) || query.trim() === '') {
+    return getDefaultQuery();
```

Comment:
I'm not sure if/when this came into play, but in the old `from_user.js` [there was an object check](https://github.com/elastic/kibana/blob/5.x/src/ui/public/parse_query/lib/from_user.js#L20-L26), and if the object was not empty it would return the object, otherwise it would return a match_all. Do we still need a similar check here?